easybuild/base/optcomplete.py

```diff
@@ -594,9 +594,8 @@ def autocomplete(parser, arg_completer=None, opt_completer=None, subcmd_complete
         if isinstance(debugfn, logging.Logger):
             debugfn.debug(txt)
         else:
-            f = open(debugfn, 'a')
-            f.write(txt)
-            f.close()
+            with open(debugfn, 'a') as f:
```

Comment:
@Flamefire We shouldn't use single-letter variables outside of list comprehensions, so please use `fh` (file handle) rather than `f`.

Bonus points if you can make `flake8` trip over this first (by tweaking `setup.cfg` so the `linting` workflow we already have will fail).

Reply:
This requires a flake8 plugin. Not doing here as maybe more will pop up, maybe open an issue to introduce that later

Reply:
Follow-up on this via #3568